lazy_tensor_core/lazy_bench.py

```diff
@@ -442,25 +442,30 @@ def lazy_compute_experiment(args, experiment, results, benchmark, lazy_benchmark
     return (speedup, pvalue)
 
 
-def check_results(correct_result, lazy_result, device):
-    # to_device has recursive logic and special handling for
-    # extracting relevant tensors from huggingface data structures
-    correct_result = to_device(correct_result, device)
-    lazy_result = to_device(lazy_result, device)
-
-    # after to_device, tuples are still tuples, but their tensors are on an eager device
+def check_results_impl(correct_result, lazy_result):
+    # recursive helper for dealing with nested data structures
     if type(correct_result) is tuple:
         for c, l in zip(correct_result, lazy_result):
-            if not torch.allclose(c, l):
-                return False
-        return True
+            return check_results_impl(c, l)
 
     if type(correct_result) is dict:
-        raise NotImplementedError("TODO(whc) missing case for dict results here; not sure if it's actually needed")
+        print(correct_result.keys())
+        for k in correct_result:
+            assert k in lazy_result
+            return check_results_impl(correct_result[k], lazy_result[k])
 
     assert type(correct_result) is torch.Tensor, f"Expect torch.Tensor but got {type(correct_result)}."
     return torch.allclose(correct_result, lazy_result)
 
+def check_results(correct_result, lazy_result, device):
+    # to_device has recursive logic and special handling for
```

Comment:
LGTM. But I wonder maybe it's better to handle huggingface data structures here too instead of `to_device`.

Reply:
yea, i think it might be better to do it here too.  i didn't want to spend the time now though.